src/compiler/proposals-and-stabilization.md

```diff
@@ -0,0 +1,336 @@
+    membership criteria, etc
+
+### Compiler flags
+
+- Adding a compiler option for internal-use only (e.g. `-Ztreat-bug-as-err`)
+  - **Propose using:** PR
+  - **Approve using:** r+
+  - If a team member finds the new option reasonable then they can merge the change adding the
+    option
+- Adding a simple compiler option with intent to later stabilize
+  - **Propose using:** MCP
+  - **Approve using:** Seconding
+  - Simple options, such as exposing an uncontroversial option from LLVM, can be implemented and
+    merged with a seconded MCP and r+ approval from a reviewer. It will need a full FCP when it is
+    later stabilized
+- Adding a complex compiler option with intent to later stabilize
+  - **Propose using:** RFC
+  - **Approve using:** FCP
+  - If the option is complicated and requires design considerations, then write and submit
+    a `t-compiler` RFC
+- Removing internal-use only flags
+  - **Propose using:** MCP
+  - **Approve using:** Seconding
+  - Describe the rationale for removing the unstable implementation. Once discussion has concluded,
+    a team member may second the proposal
+- Removing flags which were intended for eventual stabilization
+  - **Propose using:** MCP
+  - **Approve using:** Seconding
+  - Describe the rationale for removing the unstable implementation. Once discussion has concluded,
+    a team member may second the proposal
+- Stabilizing a compiler option
+  - **Propose using:** PR
+  - **Approve using:** FCP
+  - Open a PR and follow the [stabilization guide][stabilization_guide]. The assigned reviewer will
+    check that the stabilization guide has been followed, review the code and start an FCP
+- Reverting stabilization of a compiler option
+  - **Propose using:** PR
+  - **Approve using:** FCP
+  - Open a PR and follow the [stabilization guide][stabilization_guide]. The assigned reviewer will
+    check that the stabilization guide has been followed, review the code and start an FCP
+
+### Attributes
+
+- Adding a attribute for internal-use only (e.g. `rustc_attrs`)
+  - **Propose using:** PR
+  - **Approve using:** r+
+  - If a team member finds the new attribute reasonable then they can merge the change adding the
+    attribute
+- Adding a attribute with intent to later stabilize
+  - Follow the language team's process and have the implementation PR reviewed by a member of the
+    compiler team
+- Removing internal-use only attributes
+  - **Propose using:** MCP
+  - **Approve using:** Seconding
+  - Describe the rationale for removing the unstable implementation. Once discussion has concluded,
+    a team member may second the proposal
+- Removing attribute which were intended for eventual stabilization
+  - Follow the language team's process and have the removal PR reviewed by a member of the compiler
+    team
+- Stabilizing an attribute
+  - Follow the language team's process and have the stabiization PR reviewed by a member of the
+    compiler team
```

Comment:
Question: cf. https://github.com/rust-lang/rustc-dev-guide/pull/2219, may want to double-check if the attribute has any non-trivial implementation/typesystem effects that warrant T-{compiler,types} FCP in addition to T-lang FCP?

Reply:
Fixed!